token-lending/program/src/processor.rs

```diff
@@ -45,80 +60,148 @@ pub fn process_instruction(
+            collateral_amount,
+            collateral_amount_type,
+        } => {
             msg!("Instruction: Withdraw Obligation Collateral");
-            process_withdraw_obligation_collateral(program_id, collateral_amount, accounts)
+            process_withdraw_obligation_collateral(
+                program_id,
+                collateral_amount,
+                collateral_amount_type,
+                accounts,
+            )
         }
-        LendingInstruction::SetLendingMarketOwner { new_owner } => {
-            msg!("Instruction: Set Lending Market Owner");
-            process_set_lending_market_owner(program_id, new_owner, accounts)
+        LendingInstruction::BorrowObligationLiquidity {
+            liquidity_amount,
+            liquidity_amount_type,
+        } => {
+            msg!("Instruction: Borrow Obligation Liquidity");
+            process_borrow_obligation_liquidity(
+                program_id,
+                liquidity_amount,
+                liquidity_amount_type,
+                accounts,
+            )
+        }
+        LendingInstruction::RepayObligationLiquidity {
+            liquidity_amount,
+            liquidity_amount_type,
+        } => {
+            msg!("Instruction: Repay Obligation Liquidity");
+            process_repay_obligation_liquidity(
+                program_id,
+                liquidity_amount,
+                liquidity_amount_type,
+                accounts,
+            )
+        }
+        LendingInstruction::LiquidateObligation {
+            liquidity_amount,
+            liquidity_amount_type,
+        } => {
+            msg!("Instruction: Liquidate Obligation");
+            process_liquidate_obligation(
+                program_id,
+                liquidity_amount,
+                liquidity_amount_type,
+                accounts,
+            )
         }
     }
 }
 
 fn process_init_lending_market(
     program_id: &Pubkey,
-    market_owner: Pubkey,
+    lending_market_owner: Pubkey,
+    loan_to_value_ratio: u8,
+    liquidation_threshold: u8,
     accounts: &[AccountInfo],
 ) -> ProgramResult {
+    if loan_to_value_ratio >= 100 {
+        msg!("Loan to value ratio must be in range [0, 100)");
+        return Err(LendingError::InvalidConfig.into());
+    }
+    if liquidation_threshold <= loan_to_value_ratio || liquidation_threshold > 100 {
+        msg!("Liquidation threshold must be in range (LTV, 100]");
+        return Err(LendingError::InvalidConfig.into());
+    }
+
     let account_info_iter = &mut accounts.iter();
     let lending_market_info = next_account_info(account_info_iter)?;
     let quote_token_mint_info = next_account_info(account_info_iter)?;
     let rent = &Rent::from_account_info(next_account_info(account_info_iter)?)?;
     let token_program_id = next_account_info(account_info_iter)?;
 
+    assert_rent_exempt(rent, lending_market_info)?;
+    assert_uninitialized::<LendingMarket>(lending_market_info)?;
```

Comment:
Ah, interesting. So that's why it did that assignment of the fields. I can revert and do the same elsewhere, though it definitely makes it more verbose.

Reply:
I think that assignment of the fields is error-prone because it's easy to forget to initialize something, so I'm not suggesting going back to that exactly, maybe an `initialize` method would work well? Especially if it was inlined

Reply:
Good idea

Reply:
Done